guide/oauth2/README.md

```diff
@@ -0,0 +1,200 @@
+<div id='info'>
+    Hoi!
+</div>
+```
+
+You can start your server with `node index.js`. Once you start it, try connecting to [localhost](http://localhost:53134) and you should see Hoi!
+
+### Getting an OAuth2 url
+
+Now that you have your web server up and running, it's time to get some information from Discord. Head over to [Discord](https://discordapp.com/developers/applications/) and click `Create an application` where you'll be greeted with the following page.
+
+![img](/assets/img/1ch98sm.png)
+
+Take note of the `client id` field, the `client secret` field, and the `OAuth2` link on the left side of the page. For now, click on `OAuth2` and add a redirect url to `http://localhost:53134` like so.
+
+![img](/assets/img/9fejia2.png)
+
+Once you've added your redirect url, you will want to generate an OAuth2 url. Lower down on the page, you can conveniently find an OAuth2 Url Generator provided by Discord. Let's use this to generate a url for ourselves with the `identify` scope. 
+
+![img](/assets/img/18e2dwi.png)
+
+The `identify` scope will allow your application to get basic user information from Discord. A list of all scopes can be found [here](https://discordapp.com/developers/docs/topics/oauth2#shared-resources-oauth2-scopes).
+
+### Putting it together
+
+We have our website, and we have a url. Now we need to use those two things to get an access token. For basic applications like [SPAs](https://en.wikipedia.org/wiki/Single-page_application), getting an access token directly is enough to work with. If you want to do this, make sure the `response_type` in the url is `token`. However, this means you will not get a refresh token, which means the user will have to explicitly re-authorize when this access token has expired.
+
+After you change the response type, you can test the url right away. Try visiting it in your browser and you will be directed to a page that looks like this.
+
+![img](/assets/img/49jali8.png)
+
+You can see that by clicking `Authorize`, you are allowing the application to access your username and avatar. Once you click through, you should be redirected to the redirect url with `#access_token=ACCESS_TOKEN&token_type=Bearer&expires_in=EXPIRATION&scope=identify` appended to it. You now have an access token and can make requests to Discord's API to get information on the user. Let's modify `index.html` to add our OAuth2 url and to take advantage of the access token if it exists.
+
+```html
+<!DOCTYPE html>
+<div id='info'>
+	Hoi!
+</div>
+<a id='login' style='display:none;' href='your oauth2 url here'>Identify Yourself</a>
+<script>
+	window.onload = () => {
+		const match = window.location.hash.match(/access_token=(.+?)&token_type=(.+?)(?:&|$)/);
+		if (match) {
+			const [, access_token, token_type] = match;
+			const xml = new XMLHttpRequest();
+			xml.onload = () => {
+				const { username, discriminator } = JSON.parse(xml.response);
+				document.getElementById('info').innerText += `, ${username}#${discriminator}`;
```

Comment:
When you authorize the app, the text shows up as "Hoi!, User#1234", so I think you should either entirely replace the `.innerText` (to make it display "Hoi, User#1234!"), or append to it without the comma.